### DISCUSSION TITLE
What do I do wrong? Mosh has stopped working.

### DISCUSSION THREAD

**Discussion by avysk** [blinksh/blink#1815] [Q&A] [answered]
I have the following situation:

- `mosh <host>` used to work
- I have changed nothing in Blink config; however, there probably were autoupdates on iPad and on remote host (Ubuntu 23.04)
- `mosh <host>` does not work anymore: "Could not resolve address locally for host."
- `ssh <host>` works
- `mosh -2 <host>` works

What am I doing wrong?

**Comment by carloscabanero** [comment]
Do you have Host configured that is pointing to the remote IP or is there any type DNS somewhere else? If you have a Host configured, and you go to Mosh > Advanced, can you take a screenshot of the values there?

I'm looking into this because we made a patch for https://github.com/blinksh/blink/issues/1803, but the "default" type of resolution should not have changed. We have another report https://github.com/blinksh/blink/issues/1801. So I may have been the one who messed up.

Thanks

**Reply by avysk** [reply]
`<host>` is not resolvable by itself; there is a hostname set for it in blink config, and it is resolvable via DNS. I will configure resolvable host and report the results.

**Reply by avysk** [reply]
Making `<host>` resolvable allows me to connect via mosh.

**Reply by carloscabanero** [reply]
When you say "making the host resolvable", you meant you set the IP directly in Blink?

Could you please send the output of mosh --verbose so I can also see if there is any mismatch somewhere?

Trying to pin point it so we can have a fix with the new version next week.

**Reply by avysk** [reply]
I can also connect using `mosh --experimental-remote-ip=default <unresolvable alias>`. The corresponding option in config is "None".

**Reply by avysk** [reply]
No. No IP. Just using proper hostname (which DNS understands) as an alias.

**Reply by avysk** [reply]
```
blink> mosh --verbose <host>
MoshClient:DEBUG:mosh-server new -s -c 256 -l LC_ALL=en_US.UTF-8
MoshClient:DEBUG:ssh -v -o compression=no -t <host> -- mosh-server new -s -c 256 -l LC_ALL=en_US.UTF-8
socket_callback_connected: Socket connection callback: 1 (0)
ssh_key_cmp: key types don't match!
ssh_key_cmp: key types don't match!
ssh_known_hosts_read_entries: Failed to open the known_hosts file '/etc/ssh/ssh_known_hosts': No such file or directory
ssh_key_cmp: key types don't match!
ssh_key_cmp: key types don't match!
ssh_packet_userauth_failure: Access denied for 'none'. Authentication that can continue: publickey
ssh_agent_get_ident_count: Answer type: 12, expected answer: 12
ssh_packet_global_request: Invalid SSH_MSG_GLOBAL_REQUEST packet
channel_request: Channel request env failed
MoshClient:DEBUG:getaddrinfo failed with code: 8
Could not resolve address locally for host.
```

**Reply by avysk** [reply]
The ssh command from line 3 works, running mosh-server. So it is client who fails.

**Reply by carloscabanero** [reply, marked as answer]
Yep, I think you just found out what is going. The "default" option is not going through unless made explicit. Will do a hot-fix for this so it can be pushed next week.

Many thanks for the help.

**Reply by carloscabanero** [reply]
We just pushed a fix for this on 16.1.12. Should hopefully be approved soon.
